### PR TITLE
Adding to PackageIndex all of the information about our release from 6.0 and fixing packages that needed placeholders

### LINKED ISSUE #27100: Update inbox data for UAP 6
A number of assemblies shipped inbox in `release/uwp6.0` (10.0.15138), but have been rev'ed in master and ship ref-assms that would apply to UAP.

This will cause a ref-def mismatch if we don't include the implementation assembly for UAP in the package.

We need to make sure to update the packageIndex in to account for this and then ensure that the current packages validate cleanly (either rolling back reference assembly versions or adding UAP runtime assemblies to fix errors).

This **must** be done before we ship the packages again.  Ideally it's done before we ship a preview of the packages, so that we don't break folks if we decide to roll-back reference assembly versions.

/cc @joperezr @weshaggard

### PR DESCRIPTION
**Pull request by joperezr**
Fixes #27100

After shipping UAP 6.0 we didn't update the package index with this data so package validation has not been testing for breaks on packages that we will soon release. These changes include updating said index and reacting to the validation errors.

cc: @weshaggard @ericstj @joshfree

### CODE REVIEW

**Comment by joshfree** [comment]
can you PR this to master first instead of release/2.1?

File: external/uap/uap.depproj [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <NuGetDeploySourceItem>Reference</NuGetDeploySourceItem>
+    <UAPPackageVersion>6.0.7</UAPPackageVersion>
+    <TargetFramework>uap10.0.16299</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NETCore.UniversalWindowsPlatform">
+      <Version>$(UAPPackageVersion)</Version>
+    </PackageReference>
+    <PackageReference Include="System.Reflection.Emit">
```
Comment (ericstj) [inline external/uap/uap.depproj]: Please add a comment around these additions and their purpose.

File: src/System.Reflection.DispatchProxy/src/Configurations.props [hunk 2]
```diff
@@ -5,6 +5,8 @@
       netfx;
       netstandard;
       netcoreapp2.0;
+      uap10.0.16299-Windows_NT;
+      uap10.0.16299aot-Windows_NT;
       uapaot-Windows_NT;
```
Comment (ericstj) [inline src/System.Reflection.DispatchProxy/src/Configurations.props]: Following the netcoreapp2.0 pattern perhaps you should remove the version-less ones into BuildConfigurations.

File: src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj [hunk 3]
```diff
@@ -3,10 +3,12 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{7DF3C428-AAD6-41C7-98E6-6CACFD5C391E}</ProjectGuid>
+    <PackageTargetFramework>netstandard2.0;uap10.0.16299</PackageTargetFramework>
```
Comment (ericstj) [inline src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj]: Shouldn't the NS2.0 version apply or are we *still* missing a NuGet.Frameworks update?

File: external/uap/uap.depproj (hunk 1 above)
Comment (joperezr) [reply external/uap/uap.depproj]: Will do

File: src/System.Reflection.DispatchProxy/src/Configurations.props (hunk 2 above)
Comment (joperezr) [reply src/System.Reflection.DispatchProxy/src/Configurations.props]: sounds good, I'll do that.

File: src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj (hunk 3 above)
Comment (joperezr) [reply src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj]: I believe we are still missing an update of the mappings.

**Comment by joperezr** [comment]
@joshfree 
>can you PR this to master first instead of release/2.1?

That was the original idea, but because we have turned off all of the uap build in there, it was impossible to get the right validation since the builds were incomplete.

File: dir.props [hunk 4]
```diff
@@ -148,8 +148,8 @@
 
     <_packageRID/>
     <_packageRID Condition="'$(PortableBuild)' == 'true'">$(_portableOS)-$(ArchGroup)</_packageRID>
-    <_packageRID Condition="'$(TargetGroup)' == 'uap'">win10-$(ArchGroup)</_packageRID>
-    <_packageRID Condition="'$(TargetGroup)' == 'uapaot'">win10-$(ArchGroup)-aot</_packageRID>
+    <_packageRID Condition="$(TargetGroup.StartsWith('uap'))">win10-$(ArchGroup)</_packageRID>
```
Comment (weshaggard) [inline dir.props]: Why this change?

Comment (weshaggard) [reply dir.props]: Nevermind I see the other aot configurations.

File: src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj (hunk 3 above)
Comment (weshaggard) [reply src/System.Reflection.DispatchProxy/ref/System.Reflection.DispatchProxy.csproj]: I would expect this mapping to exist as well.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 5]
```diff
@@ -811,11 +826,13 @@
         "4.2.1.0": "4.5.0"
       }
     },
+    /*The shim of System.ComponentModel.Composition shipped on netcoreapp2.0 and uap6.0 and we now have an implementation
```
Comment (weshaggard) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Would you mind fixing this while you are in here.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 6]
```diff
@@ -832,6 +832,8 @@
       "InboxOn": {
         "monoandroid10": "Any",
         "monotouch10": "Any",
+        "netcoreapp2.0": "Any",
```
Comment (weshaggard) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: You can remove the comment now.

**Comment by joperezr** [comment]
@weshaggard & @ericstj do you mind taking another peak at this before merging?

File: pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj [hunk 7]
```diff
@@ -49,6 +49,12 @@
         System.Security.Principal.Windows;
       </Value>
     </ValidatePackageSuppression>
+    <!-- Permit missing Inbox since we this used to be a shim and it is now OOB -->
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]: typo: since we this -> since this

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj [hunk 8]
```diff
@@ -66,6 +66,12 @@
         System.Private.Reflection.Metadata.Ecma335;
       </Value>
     </ValidatePackageSuppression>
+    <!-- Permit missing Inbox since we this used to be a shim and it is now OOB -->
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: typo: since we this -> since this

File: src/Microsoft.Win32.Registry/src/Configurations.props [hunk 9]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <PackageConfigurations>
       netstandard;
+      netstandard-Windows_NT;
```
Comment (weshaggard) [inline src/Microsoft.Win32.Registry/src/Configurations.props]: Why do we need this netstandard-Windows_NT configuration?

File: src/System.IO.FileSystem.AccessControl/src/Configurations.props [hunk 10]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
   <PackageConfigurations>
     netstandard;
+    netstandard-Windows_NT;
```
Comment (weshaggard) [inline src/System.IO.FileSystem.AccessControl/src/Configurations.props]: Why do we need this configuration?

File: src/System.ServiceProcess.ServiceController/src/Configurations.props [hunk 11]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <PackageConfigurations>
       netstandard;
+      netstandard-Windows_NT;
```
Comment (weshaggard) [inline src/System.ServiceProcess.ServiceController/src/Configurations.props]: Why is this needed? In general I like to avoid RID netstandard configurations.

**Comment by weshaggard** [comment]
@joperezr we are going to be merging master into release/2.1 later this week. It might be worth waiting on this until after that otherwise I expect the merge is going to be a little challenging.

File: pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj (hunk 7 above)
Comment (joperezr) [reply pkg/Microsoft.Private.CoreFx.NETCoreApp/Microsoft.Private.CoreFx.NETCoreApp.pkgproj]: Fixed.

File: src/Microsoft.Win32.Registry/src/Configurations.props (hunk 9 above)
Comment (joperezr) [reply src/Microsoft.Win32.Registry/src/Configurations.props]: Because since we harvest a Windows specific netstandard1.3, then package validation will complain given that the compile asset resolved for UAP6.0 (the netstandard2.0 ref) will have a higher assembly version than the resolved implementation for UAP6.0 (the windows specific netstandard1.3 implementation) because the runtime specific assets will be prefered over the non rid specific (lib/netstandard2.0) . This is required in order to pass package validation and in order to be consumed on UAP.

File: src/System.IO.FileSystem.AccessControl/src/Configurations.props (hunk 10 above)
Comment (joperezr) [reply src/System.IO.FileSystem.AccessControl/src/Configurations.props]: Same reason as above.

File: src/System.ServiceProcess.ServiceController/src/Configurations.props (hunk 11 above)
Comment (joperezr) [reply src/System.ServiceProcess.ServiceController/src/Configurations.props]: I agree, but as explained above, this is required because we are harvesting a windows specific netstandard1.3 implementation for this package, which will be prefered over the lib/netstandard2.0 implementation for UAP.

**Comment by joperezr** [comment]
sounds good, I'll wait for the merge from master and then I'll merge this.

File: src/Microsoft.Win32.Registry/src/Configurations.props (hunk 9 above)
Comment (weshaggard) [reply src/Microsoft.Win32.Registry/src/Configurations.props]: @ericstj do you think adding the netstandard-Windows_NT configuration is better then remapping the harvested location to put the NS13 asset in the lib instead of the windows RID. I suppose there is a chance that remapping the path might actually cause breaks but it would depend on the actual package.

Comment (ericstj) [reply src/Microsoft.Win32.Registry/src/Configurations.props]: That ns1.3 asset is a *windows specific implementation*.  If you started putting that in a RID-less folder you'd change the behavior for non-windows (previously it got a PNS exception).

If you go back and look at the history as to why this is targeting NETCoreapp-window_nt today instead of netstandard-windows_nt, it is because of this: https://github.com/joperezr/corefx/commit/28097ca87cc2e8bcf7760f05732464b2c00eeaa4#diff-6b3a566cbc8f4a17c79fa82c0a0feafc.  I needed to create a netcoreapp configuration to avoid an assembly cycle in the implementation of netcoreapp.

In that case the better fix here would be to move the netcoreapp-* configurations into BuildConfigurations only (since their only difference from netstandard version will be referenced assemblies) and have a netstandard-windows_NT version in the package.

File: src/System.ServiceProcess.ServiceController/src/Configurations.props (hunk 11 above)
Comment (ericstj) [reply src/System.ServiceProcess.ServiceController/src/Configurations.props]: @weshaggard why do you want to avoid RID-netstandard configurations?  What if we want to share a windows implementation between netcoreapp and UAP?  That's the reason why those old harvested assets existed that were RID-specific and targeted netstandard.

Comment (ericstj) [reply src/System.ServiceProcess.ServiceController/src/Configurations.props]: There is nothing impure about having a RID + netstandard assets.  Consider that the netstandard version targeted by the assembly represents the managed API needed by that assembly and the RID represents the platform API/behavior.  It's actually more pure to have the assembly be win+netstandard since that's technically all it depends on.  Making it win+UAP / win+netcoreapp implies it depends on something specific to those TFMs which at least the package asset does not.

Comment (weshaggard) [reply src/System.ServiceProcess.ServiceController/src/Configurations.props]: By purity I'm thinking about netstandard being pure IL and no pinvokes, which is where I would like most libraries to live. I know that isn't a reality but it is my ideal world. I agree using a combination of netstandard+RID can be useful in some scenarios but it also adds some risk as the native dependencies may or may not work on any netstandard supported platform. I guess that is true for all assets that go under a RID and I've never really liked how "undefined" RID's are which also leads me to not liking these. At any rate netstandard+RID is a tool and we should use it if necessary but that doesn't make me like it :)

**Comment by ChadNedzlek** [comment]
Is this making progress? 9 days ago we said "later this week", but seems like there hasn't been any activity recently. This is blocking a buildtools uptake which is blocking some infrastructure work.

**Comment by ericstj** [comment]
I think @joperezr mentioned that he was going to undo the buildtools change in release and wait for the changes from master to integrate.  @joperezr are you still planning on that?

**Comment by joperezr** [comment]
yup already did that. This work has been now moved to master and merged so I'm closing this PR now.
